### ISSUE TITLE
Update/clarify code convention

### ISSUE THREAD

**Issue by jcohenadad** [shimming-toolbox/shimming-toolbox-matlab#70]
See for example:
- http://www.cosmomvpa.org/contribute.html#matlab-code-guidelines

**Comment by rtopfer** [comment]
already there, top of [CONTRIBUTING.md](https://github.com/shimming-toolbox/shimming-toolbox/blob/master/CONTRIBUTING.md#programming-guidelines)

**Comment by jcohenadad** [comment]
this line:
```
When contributing to the library, for maintainability, please follow the programming guidelines for Matlab by Richard Johnson.
```
redirects to a mathworks page, which requires a username/passwd to be able to download a zip file, which needs unpacking, to finally get to the pdf which is 43 pages.

It is much more common and intuitive to just get the guidelines written *inside* the contribution guidelines (not just a link to it). It notably enables to directly point to a feature of the guideline (e.g. arguments should be written that way) when doing code review.

Moreover, this pdf is from 2014, and new features (classes, block argument definition) are not taken into account.

**Comment by rtopfer** [comment]
It's an extremely useful document + remains relevant. you can read it in 5-10 minutes: 43 pages w/ huge font + mostly white space.

doesn't cover everything + i agree there need to be other points addressed in Contributing. but this pdf should still be mandatory reading as far as I'm concerned.

**Comment by jcohenadad** [comment]
> It's an extremely useful document + remains relevant. you can read it in 5-10 minutes: 43 pages w/ huge font + mostly white space.

agreed

> doesn't cover everything + i agree there need to be other points addressed in Contributing. but this pdf should still be mandatory reading as far as I'm concerned.

i've updated the title to be more precise

**Comment by rtopfer** [comment]
@jcohenadad do you want to adopt the `do_this_thing()` style for MATLAB functions/methods from now on? unconventional for Matlab, though it's clearly more readable than `dothisthing()`. previously i've reserved underscores for constants, but these are also indicated with all caps, e.g. `DEFAULT_GYRO=42.576E6`.

non-constant variables are always camelCase, but i've tended to use a leading capital for struct/object/cell variables to indicate that they contain more stuff than simple types. However idk that I actually want to stick with that practice. @kousu what do you think? lowercase for all variables? 

@jcohenadad i had started drafting a bunch of material to add to the Contributing page a couple weeks ago before realizing I was just wasting my time rewriting what's already covered in the [style guide by Richard Johnson](https://www.mathworks.com/matlabcentral/fileexchange/46056-matlab-style-guidelines-2-0)... Should we just have a brief bullet summary in Contributing.md along with the .pdf link?

**Comment by jcohenadad** [comment]
> @jcohenadad do you want to adopt the `do_this_thing()` style for MATLAB functions/methods from now on? unconventional for Matlab, though it's clearly more readable than `dothisthing()`. previously i've reserved underscores for constants, but these are also indicated with all caps, e.g. `DEFAULT_GYRO=42.576E6`.
> non-constant variables are always camelCase, but i've tended to use a leading capital for struct/object/cell variables to indicate that they contain more stuff than simple types. However idk that I actually want to stick with that practice. @kousu what do you think? lowercase for all variables?

right, global (immutable) variables are traditionally all caps snake case (as you wrote above). The reason i went with the snake_case convention is because i noticed you were using camelCase for variables, so i wanted to make a clear distinction between variables and functions. Regarding classes, i've noticed a consensus towards PascalCase (as you also adopted).

> @jcohenadad i had started drafting a bunch of material to add to the Contributing page a couple weeks ago before realizing I was just wasting my time rewriting what's already covered in the [style guide by Richard Johnson](https://www.mathworks.com/matlabcentral/fileexchange/46056-matlab-style-guidelines-2-0)... Should we just have a brief bullet summary in Contributing.md along with the .pdf link?

that sounds good to me!

**Comment by rtopfer** [comment]
Re: PascalCase, I gather it's conventional for class _definitions_ but I've tended to use it for object instances (and structs, cells) as well (e.g. `Field = FieldEval()`), which I think might be unusual? It's reminiscent of older languages like FORTRAN that prefix variable names with the type. (I've noticed Siemens still does this sometimes in their C++ code.)

Wondering if it should be adopted or discouraged in the new guidelines.

**Comment by jcohenadad** [comment]
> Re: PascalCase, I gather it's conventional for class _definitions_ but I've tended to use it for object instances (and structs, cells) as well (e.g. `Field = FieldEval()`), which I think might be unusual? It's reminiscent of older languages like FORTRAN that prefix variable names with the type. (I've noticed Siemens still does this sometimes in their C++ code.)
> 
> Wondering if it should be adopted or discouraged in the new guidelines.

i would treat these object instances as variables, so camelCase.
